src/core/hooks/request-context/request-service.ts

```diff
@@ -85,39 +109,91 @@ class RequestService {
             return;
         }
 
-        this._setters.setRequests((prev) => {
-            if (index < 0 || index > prev.length) {
+        this._setter((prev) => {
+            if (index < 0 || index > prev.requests.length) {
                 console.error(
                     `Error removing request with index "${indexString}". Index out of bounds`,
                 );
                 return prev;
             }
 
-            const next = [...prev.slice(0, index), ...prev.slice(index + 1)];
+            const next: RequestServiceState = {
+                ...prev,
+                requests: [
+                    ...prev.requests.slice(0, index),
+                    ...prev.requests.slice(index + 1),
+                ],
+            };
+
+            if (prev.currentRequest?.id === prev.requests[index].id) {
+                // Current removed
+
+                const nextCurrentIndex = prev.requests.findIndex(
+                    (item) => item.id === prev.currentRequest?.id,
+                );
 
-            if (this.currentRequestIndex ?? index >= next.length) {
-                this._setters.setCurrentRequest(next[next.length - 1]);
+                if (nextCurrentIndex >= next.requests.length) {
+                    next.currentRequest =
+                        next.requests[next.requests.length - 1];
+                } else {
+                    next.currentRequest = next.requests[nextCurrentIndex];
+                }
             }
 
             return next;
         });
     }
 
-    async save(): Promise<string> {
+    /** Save current workspace.
+     * @param filepath If not defined then it will save to internal app state.
+     */
+    async save(filepath?: string): Promise<string> {
+        if (filepath) {
+            this._setter((prev) => ({
+                ...prev,
+                filepath,
+            }));
+        }
+
         return invoke<string>('save_app_state', {
             state: JSON.stringify({
                 currentRequest: this.#currentRequest,
+                filepath: this.#filepath,
```

Comment:
We are passing `this.#filepath` but we just set the state so we may be using old version.

Reply:
Fixed.